### ISSUE TITLE
[Bug]: Design system dropdown fixes

### ISSUE THREAD

**Issue by albinAppsmith** [appsmithorg/appsmith#16997]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

1. For single select dropdown keyboard navigation does not work if not option is selected.
2. For single select dropdown if an option is already selected and the we try to press the up/down arrow keys, the selected option disappears and we cannot navigate further.
3. For multi select dropdown clicking Enter does not select the options.
4. For multi select dropdown there is an infinite scroll i.e. if we press down arrow key after the last option, the first option does not get selected rather the highlight disappears.
5. There is a padding for dropdown option wrapper in the bottom.

### Steps To Reproduce

1. Check dropdown in storybook with different props
or
1. Check dropdown behaviours in Invite modal, G-sheet 

### Version

@appsmithorg/design-system-1.0.24
